src/components/User/SystemAssets/components/AssetList.vue

```diff
@@ -60,28 +93,46 @@ export default {
     }
   },
 
-  created () {
-    this.getAssets()
-  },
-
   methods: {
     async getAssets () {
       this.assets = []
       this.$store.commit('OPEN_LOADER')
+      this.isLoading = true
+      let response = {}
       try {
-        const response = await Sdk.horizon.assets.getAll({
-          owner: config.MASTER_ACCOUNT
-        })
-        this.assets = response.data.map(item => {
-          const creatorDetails = item.details || item.creatorDetails
-          return Object.assign(item, { creatorDetails })
-        })
+        response = await ApiCallerFactory
+          .createCallerInstance()
+          .getWithSignature('/v3/assets', {
+            filter: { owner: config.MASTER_ACCOUNT }
+          })
         this.$store.commit('CLOSE_LOADER')
       } catch (err) {
         console.error('caught error', err)
         this.$store.commit('CLOSE_LOADER')
         this.$store.dispatch('SET_ERROR', 'Something went wrong. Can\'t to load assets list')
       }
+      this.isLoading = false
+      return response
+    },
+
+    setAssets (data) {
+      this.assets = this.addСreatorDetailsKeyToObjects(data)
+    },
+
+    async extendAssets (data) {
+      try {
+        const mapedData = this.addСreatorDetailsKeyToObjects(data)
+        this.assets = this.assets.concat(mapedData)
+      } catch (error) {
+        ErrorHandler.process(error)
+      }
+    },
+
+    addСreatorDetailsKeyToObjects (array) {
+      return array.map(item => {
+        const creatorDetails = item.details || item.creatorDetails
+        return Object.assign(item, { creatorDetails })
+      })
```

Comment:
presence of both `details` and `creatorDetails` are likely bug, only `creatorDetails` should exist. We should inform back-end about it, please prepare steps to reproduce the error and screens.

The function is generally normalizes the used data, thus can be (and recommended) to rename into something like "normalizeAssetsCreatorDetails"

Reply:
I have rechecked response list items and in all items there is no any `creatorDetails` field. Is there a necessity for creating bug report or I just can remove normalization function and replace all uses of `creatorDetails` field to `details`?